lib/features/autentication/login_page.dart

```diff
@@ -1,45 +1,38 @@
 import 'package:flutter/material.dart';
-import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:go_router/go_router.dart';
+import 'package:moneyapp/common/cubit/login_cubit.dart';
 import 'package:moneyapp/router/go_router.dart';
 
-class LoginPage extends StatefulWidget {
+class LoginPage extends HookWidget {
   const LoginPage({super.key});
 
-  @override
-  LoginPageState createState() => LoginPageState();
-}
-
-class LoginPageState extends State<LoginPage> {
-  final TextEditingController _emailController = TextEditingController();
-  final TextEditingController _passwordController = TextEditingController();
-
-  final GlobalKey<ScaffoldMessengerState> _scaffoldMessengerKey =
-      GlobalKey<ScaffoldMessengerState>();
-
-  Future<void> _login() async {
+  Future<void> _login(
+      BuildContext context,
+      LoginCubit cubit,
+      TextEditingController emailController,
+      TextEditingController passwordController) async {
     try {
-      await FirebaseAuth.instance.signInWithEmailAndPassword(
-        email: _emailController.text,
-        password: _passwordController.text,
-      );
-
-      if (mounted) {
-        AppRouter.router.go('/');
-      }
+      await cubit.login(emailController.text, passwordController.text);
+      AppRouter.router.go('/');
     } catch (e) {
-      _scaffoldMessengerKey.currentState?.showSnackBar(
-        const SnackBar(content: Text('Login failed. Please try again.')),
+      ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(content: Text('Login failed: ${e.toString()}')),
```

Comment:
Use BlocListener (or BlocConsumer) to ilsten to AuthError state being emitted. If it is emitted, show the snackbar.